project_setup.py

```diff
@@ -0,0 +1,240 @@
+
+    Args:
+        user_provided_repo_url (str, optional): The URL of the remote repository. Defaults to "".
+
+    Returns:
+        str: The URL of the remote repository.
+
+    """
+    url_guess = subprocess.check_output(
+        ['git', 'remote', 'get-url', 'origin']
+    ).decode().strip()
+
+    return user_provided_repo_url or url_guess
+
+
+def get_package_name(
+    repo_url:str,
+    user_provided_package_name:str = "",
+) -> str:
+    """Gets the package name from the user.
+
+    Guesses the package name from the repo URL.
+
+    Args:
+        repo_url (str): The URL of the remote repository.
+        user_provided_package_name (str, optional): The package name provided by the user. Defaults to "".
+
+    Returns:
+        str: The package name.
+    """
+    guessed_name = make_name_safe(Path(repo_url).stem)
+    return user_provided_package_name or guessed_name
+
+
+def replace_file_contents(
+    new_str:str,
+    old_str:str,
+    file_path:str,
+) -> bool:
+    """Replaces instances of `old_str` with `new_str` in file at file_path.
+
+    Args:
+        old_str (str): The string to replace.
+        new_str (str): The string to replace `old_str` with.
+        file_path (str): The path to the file to modify.
+
+    Returns:
+        None
+    """
+    with open(file_path, "r") as f:
+        content = f.read()
+
+    new_content = content.replace(old_str, new_str)
+
+    if new_content != content:
+        with open(file_path, "w") as f:
+            f.write(new_content)
+        return True
+
+    return False
+
+
+def self_destruct() -> None:
+    """Deletes this file."""
+    os.remove(__file__)
+
+
+def rtd_project_guess(
+    repo_url:str,
+    user_provided_rtd_name:str = "",
+) -> str:
+    """Guesses the RTD project name from the repo URL."""
+    return user_provided_rtd_name or Path(repo_url).stem
+
+
+def codecov_project_guess(
+    repo_url:str,
+    user_provided_codecov_name:str = "",
+) -> str:
+    """Guesses the codecov project name from the repo URL."""
+    guess = "/".join(Path(repo_url).parts[-2:]).replace(".git", "")
+    return user_provided_codecov_name or guess
+
+
+def run_setup(
+    repo_url:str,
+    package_name:str,
+    dir_path:str = ".",
+    input_func:Callable[[Optional[str]], str] = input,
+    output_func:Callable[[str], None] = print,
+) -> None:
+    # Update pyproject.toml and README.md, which contain the repo URL===========
+    rtd_guess = rtd_project_guess(repo_url)
+    codecov_guess = codecov_project_guess(repo_url)
+
+    replace_args = zip(
+        [repo_url, repo_url, rtd_guess, codecov_guess],
+        [TEMPLATE_REPO_URL, TEMPLATE_REPO_URL,  "ssec-jhu-base-template", "ssec-jhu/base-template"],
+        map(lambda f: os.path.join(dir_path, f), ["pyproject.toml", "README.md", "README.md", "README.md"]),
+        [UPDATED_PYPROJECT_TOML, UPDATED_README_CICD, UPDATED_README_RTD, UPDATED_README_CODECOV]
+    )
+
+    for new_str, old_str, file_path, msg in replace_args:
```

Comment:
See the above comment about passing str replacements as single dict. Then, this is just looping over files, especially since the replacements are global replace-alls and not file-specific.

Reply:
I don't mind changing the iterable to files rather than terms.